clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10603,8 +10681,22 @@ static void getTripleBasedSPIRVTransOpts(Compilation &C,
               ",+SPV_KHR_non_semantic_info"
               ",+SPV_KHR_cooperative_matrix"
               ",+SPV_EXT_shader_atomic_float16_add";
-  if (IsCPU)
+  auto hasNoOffloadFP32PrecOption = [](const llvm::opt::ArgList &TCArgs) {
+    return !TCArgs.hasFlag(options::OPT_foffload_fp32_prec_sqrt,
+                           options::OPT_fno_offload_fp32_prec_sqrt, false) &&
+           !TCArgs.hasFlag(options::OPT_foffload_fp32_prec_div,
+                           options::OPT_fno_offload_fp32_prec_div, false);
+  };
+  auto shouldUseOffloadFP32PrecOption = [](const llvm::opt::ArgList &TCArgs) {
+    return (TCArgs.hasFlag(options::OPT_fno_offload_fp32_prec_sqrt,
+                           options::OPT_foffload_fp32_prec_sqrt, false) ||
+            TCArgs.hasFlag(options::OPT_fno_offload_fp32_prec_div,
+                           options::OPT_foffload_fp32_prec_div, false));
+  };
+  if (IsCPU && hasNoOffloadFP32PrecOption(TCArgs) ||
+      !IsCPU && shouldUseOffloadFP32PrecOption(TCArgs)) {
```

Comment:
```suggestion
  if ((IsCPU && hasNoOffloadFP32PrecOption(TCArgs)) ||
      shouldUseOffloadFP32PrecOption(TCArgs)) {
```
I believe the option settings should always trigger regardless if doing AOT for CPU.